### PR TITLE
Update mocking.md

### PR DESCRIPTION
**Pull request by rscircus**
Add a little bit of reasoning why it makes sense to have a configurable sleeper.

### CODE REVIEW

**Comment by rscircus** [comment]
Actually I got lost a little bit in this section and had to move the `ConfigurableSleeper` after `cleanup and refactor` to the main file. That's why expanded this section a bit in this proposal.

One might also point out that this generalization of the Sleeper doesn't break the other tests... what do you think?

**Comment by rscircus** [comment]
That grammar checker in Travis is awesome.

**Comment by rscircus** [comment]
Remark on Encapsulation & Generalization http://www.vias.org/javacourse/chap06_06.html - Downey's Books are a treat. :)

File: mocking.md [hunk 1]
```diff
@@ -480,7 +480,7 @@ We now have our function and its 2 important properties properly tested.
 
 ## Extending Sleeper to be configurable
 
-A nice feature would be for the `Sleeper` to be configurable.
+A nice feature would be for the `Sleeper` to be configurable. This means that we would like to adjust the sleep time in our main program.
```
Comment (quii) [inline mocking.md]: > This means that we would like to adjust the sleep time in our main program.

Maybe it should say

> This means that we can adjust the sleep time in our main program.

File: mocking.md [hunk 2]
```diff
@@ -568,7 +568,7 @@ func main() {
 
 If we run the tests and the program manually, we can see that all the behavior remains the same.
 
-Since we are using the `ConfigurableSleeper`,  it is safe to delete the `DefaultSleeper` implementation. Wrapping up our program.
+Since we are using the `ConfigurableSleeper`, it is now safe to delete the `DefaultSleeper` implementation. Wrapping up our program and having a more general Sleeper with arbitrary long countdowns. Adding parameters in such a fashion is called generalization as it makes entities more general and is often followed by encapsulation, where functionality is encapsulated in a function or other entity.
```
Comment (quii) [inline mocking.md]: I'm not sure how helpful this sentence is, it seems to cover a lot of concepts quickly. I'd prefer to remove that.

Comment (rscircus) [reply mocking.md]: Thanks for reviewing. :) The thing is, the whole chapter covers a large amount of topics, that's why I felt lost a bit and tried to elaborate on the topics covered.

What do you think about referring to Downey's literature the way you did this with Uncle Bobs? I'll make a suggestion later, what I mean.

Comment (quii) [reply mocking.md]: I guess I dont see how encapsulation strongly relates to what we're doing (i can kinda see it, but not sure it adds much). I'll check out your changes when you do them :)

Comment (rscircus) [reply mocking.md]: Really liked the cake example. It just nails it.

**Pull request merged**